### PR TITLE
Support for FramePort and one .ino example.

### PR DESCRIPTION
**Pull request by clavisound**
As the title says. I hope I made it properly.	

`Frame_Port` is `uint8_t` because I tested with TTN and the range was 0-255. I think normally is uint16_t? In this case inform me to modify accordingly.

I hope setPower to follow soon. I am little confused :-)

### CODE REVIEW

File: examples/hello_LoRa_frame_port/hello_LoRa_frame_port.ino [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    Serial.println("Check your radio");
+    while(true);
+  }
+  Serial.println("OK");
+}
+
+void loop()
+{
+  Serial.println("Sending LoRa Data...");
+  lora.sendData(loraData, sizeof(loraData), lora.frameCounter, framePort); // If you ommit FramePort it's assumed #1
```
Comment (brentru) [inline examples/hello_LoRa_frame_port/hello_LoRa_frame_port.ino]: Could you explain this comment? I'm not sure what you mean..
> // If you ommit FramePort it's assumed #1

**Review by brentru — CHANGES_REQUESTED** [review]
Looks good - tiny change.

Instead of making a separate `ino` example, add a commented out line to the existing example showing how you can set your own framePort, but retain the original function call.

```
lora.sendData(loraData, sizeof(loraData), lora.frameCounter);
// Optionally set the Frame Port (1 to 255)
// lora.sendData(loraData, sizeof(loraData), lora.frameCounter, framePort);
````

File: examples/hello_LoRa_frame_port/hello_LoRa_frame_port.ino (hunk 1 above)
Comment (clavisound) [reply examples/hello_LoRa_frame_port/hello_LoRa_frame_port.ino]: You described correctly the behavior in your first comment https://github.com/adafruit/TinyLoRa/pull/27#pullrequestreview-282257078 . Will make the changes.

In my testing, TTN also accepts port 0. Is out of spec?

**Comment by clavisound** [comment]
@brentru are we ok? I have also ready the setPower function if we finish with this.

**Comment by brentru** [comment]
@clavisound  I'm going to test this tomorrow AM with the TTN Gateway at work, will merge if it's OK.

File: examples/hello_LoRa/hello_LoRa.ino [hunk 2]
```diff
@@ -8,6 +8,7 @@
 // Copyright 2015, 2016 Ideetron B.V.
 //
 // Modified by Brent Rubell for Adafruit Industries, 2018
+// Added FramePort by clavisound for Adafruit Industries, Aug 2019
```
Comment (brentru) [inline examples/hello_LoRa/hello_LoRa.ino]: Your contribution will be reflected in release notes, please don't add to the pre-existing license.

File: examples/hello_LoRa/hello_LoRa.ino [hunk 3]
```diff
@@ -31,6 +32,9 @@ unsigned char loraData[11] = {"hello LoRa"};
 // How many times data transfer should occur, in seconds
 const unsigned int sendInterval = 30;
 
+// In which port we send data? [TTN supports 0-255, 1 is default]
+uint8_t framePort = 1;
```
Comment (brentru) [inline examples/hello_LoRa/hello_LoRa.ino]: `framePort` is optionally set on Line 72/73. This should be moved to the `sendData` on Line 73:

```
  // Optionally set the Frame Port (1 to 255)
  // uint8_t framePort = 1;
  // lora.sendData(loraData, sizeof(loraData), lora.frameCounter, framePort);

```

File: TinyLoRa.cpp [hunk 4]
```diff
@@ -500,9 +500,11 @@ uint8_t TinyLoRa::RFM_Read(uint8_t RFM_Address) {
               Frame counter for transfer frames.
     @param    Data_Length
               Length of data to be sent.
+    @param    Frame_Port
+              Number of frame port
```
Comment (brentru) [inline TinyLoRa.cpp]: Let's make this more verbose:

```
@param Frame_Port
                Frame port to send data from, from 0 to 225.
```

**Review by brentru — CHANGES_REQUESTED** [review]
@clavisound Works excellent with the TTN gateway in my home.

Requesting a few final changes, I'll merge in once they're added.

**Review by brentru — APPROVED** [review]
Looks good @clavisound, will merge and release when travis finishes. 

Could you bump the library version (https://github.com/adafruit/TinyLoRa/blob/master/library.properties#L2) to `1.2.0` to prepare for a release?

**Comment by clavisound** [comment]
Thanks brentru.

I have plenty of comments for the next (setPower) commit. Do you prefer them in the next PR, or in the adafruit forum?

**Comment by brentru** [comment]
@clavisound  Open a new issue on this repo and tag me in it. You'll reference the issue when the PR is submitted.

Thanks for your work on this pull request, looks great!

**Pull request merged**